### PR TITLE
Detail more precisely the two major categories of non-GCed data

### LINKED ISSUE #3437: Document what GC will *not* remove
Add to documentation all kinds of data that will not be removed by GC:

* Uncommitted
* HEADs on *any* branch
* Etc.

Try to ensure no surprises.

### PR DESCRIPTION
**Pull request by arielshaqed**
Fixes #3437.

### CODE REVIEW

File: docs/reference/garbage-collection.md [hunk 1]
```diff
@@ -90,6 +90,16 @@ The garbage collection process proceeds in two main phases:
...
   gateway will return HTTP status 410 ("Gone").
 
+### What does _not_ get collected
+
+From the above definition of what gets collected, some objects will _not_ be
+collected regardless of configured GC rules:
+
+* Any object that was _uploaded but never committed_ cannot be collected.  See
+  [#1933](https://github.com/treeverse/lakeFS/issues/1933) for more details.
+* Any object that is a branch HEAD is visible on that branch, and will not
```
Comment (johnnyaug) [inline docs/reference/garbage-collection.md:100]: Something about this sentence doesn't make sense

**Review by johnnyaug — APPROVED** [review]
Thanks! See comment

File: docs/reference/garbage-collection.md (hunk 1 above)
Comment (arielshaqed) [reply docs/reference/garbage-collection.md]: You're right!  I changed it, hopefully for the better.

**Review by arielshaqed — COMMENTED** [review]
Thanks for catching very poor phrasing!

Pulling, and we can always make it even better. :-)

**Comment by arielshaqed** [comment]
Pulling without rerunning tests, this is a docs-only change.

**Pull request merged**